docs/proposals/web-hook.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Dynamic Admission Control
+status: Alpha
+authors:
+    - "@chendave"
+approvers:
+  - TBD
+creation-date: 2019-07-19
+last-updated: 2019-07-19
+---
+
+# Motivation
+As the evolving of the project, it is foreseeable more API and resouce will be added in the project. kubeedge so far lack of effective way of pre-processing for the object configuration, for example, whether the pod that is going to be created contains the unwanted label, could be the specific configmap be protected from deletion etc.
+
+There is a concrete example on the github [issue 845](https://github.com/kubeedge/kubeedge/issues/845), the issue there cannot be addressed by [CRD validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation), so we need to explore approach to achieve that purpose.
+
+
+## Goals
+* A framework to enable admission control in kubeedge.
+* Address the edge cases where kubernetes CRD validation cannot accomplish.
+* Create basic integration testcases.
+
```

Comment:
Can we specify some non-goals here ? What problems are we not trying to solve with this approach ?